### PR TITLE
Allow setNewOrder() to accept a custom sort column

### PR DESCRIPTION
**Pull request by RalphMRivera**
e.g., `MyModel::setNewOrder(['09b25ab5-b921-4794-941c-2152982c24a6', 'ca11be8f-6faf-4848-bb9b-fc951193a66d', 
 '09b25ab5-b921-4794-951c-2152982c24a6', 'eee4041c-4aa2-468b-bga0-82d70468409d', '2643003e-2335-4496-ba4d-df6b24caa551'], 10, 'uid');`

### CODE REVIEW

**Comment by freekmurze** [comment]
Could you add tests and update the readme?

**Comment by RalphMRivera** [comment]
Hey @freekmurze I added a new method and updated the docs. However, I'm really not good with writing tests. Is there anyone that can help me with that?

**Comment by RalphMRivera** [comment]
I got someone to help out with this. :)

**Comment by joelclermont** [comment]
I'll write some tests and update the PR.

**Comment by joelclermont** [comment]
I followed the existing tests around "set new order" and created a new version for using a custom column. I didn't replicate every single test, but if you feel more coverage is needed, let me know.

File: tests/TestCase.php [hunk 1]
```diff
@@ -44,11 +44,15 @@ protected function setUpDatabase()
...
             $table->string('name');
+            $table->string('custom_column_sort');
             $table->integer('order_column');
         });
 
         collect(range(1, 20))->each(function (int $i) {
-            Dummy::create(['name' => $i]);
+            Dummy::create([
+                'name' => $i,
+                'custom_column_sort' => chr(123 - $i), // backwards from z
```
Comment (freekmurze) [inline tests/TestCase.php]: What is this voodoo? 😀 Can it be written in a more expressive way?

Comment (joelclermont) [reply tests/TestCase.php]: The goal was to have values that were not sorted the same way as the name/index/default sort, so I chose to enumerate characters backwards from the letter `z`. I could also have picked negative numbers and used `0 - $i`. Or, I could also generate a random number with `rand()`. Do you have a preference or alternate suggestion?

Comment (freekmurze) [reply tests/TestCase.php]: Let's just go with `rand`, that's a bit more expressive in this situation.

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**